pkg/provider/prefix/provider.go

```diff
@@ -296,11 +296,16 @@ func (p *ipv4PrefixProvider) CreateIPv4PrefixAndUpdatePool(job *worker.WarmPoolJ
 
 	if err != nil {
 		p.log.Error(err, "failed to create all/some of the IPv4 prefixes", "created resources", resources)
-		didSucceed = false
+
 		//TODO: This adds a dependency on EC2 API error. Refactor later.
 		if strings.HasPrefix(err.Error(), utils.InsufficientCidrBlocksReason) {
+			// Send node event to inform user of insufficient CIDR blocks error
 			utils.SendNodeEvent(p.apiWrapper.K8sAPI, job.NodeName, utils.InsufficientCidrBlocksReason,
 				utils.ErrInsufficientCidrBlocks.Error(), v1.EventTypeWarning, p.log)
+		} else {
+			// For all other errors, set didSucceed as false to re-sync the pool.
+			// However, since InsufficientCidrBlocks error is not retryable, do not set didSucceed to false.
```

Comment:
Makes sense here. But the `didSucceed` will confuse people in the future if it hits not retryable error but `didSucceed` == true...
Does that make sense to change `didSucceed` to `retry`? Or maybe for now, to avoid changing downstream code, we can rename it to `notRetry` which is true for not-retryable error but false to others?

Reply:
Sounds good, renamed it to `notRetry` for now.